src/tools/illink/src/ILLink.Shared/DataFlow/ValueSet.cs

```diff
@@ -205,6 +207,24 @@ internal static ValueSet<TValue> Meet (ValueSet<TValue> left, ValueSet<TValue> r
 			return new ValueSet<TValue> (values);
 		}
 
+		internal static ValueSet<TValue> Intersection (ValueSet<TValue> left, ValueSet<TValue> right)
+		{
+			if (left._values == null)
+				return left;
```

Comment:
Might be worth adding a comment that we don't need DeepCopy, since the value is empty. (It's confusing that Union has to call DeepCopy everythwere, and Intersetion only sometimes)
Or alternatively we migth return `Empty` - for better readability.